File: Array/fanbei/0623/java/Solution06_23.java
package com.example.lsn37_dagger2_subcomponent;

public class Solution06_23 {

    public int findNumbers(int[] nums){
        int count = nums.length;
        if(count == 0 ){
            return 0;
        }
        int result = 0;
        int tmp = 0,bit = 0;
        for (int i = 0;i<count;i++){
            tmp = nums[i];
            bit = 0;
            while (tmp != 0) {
                tmp = tmp/10;
                bit++;
            }
            if (bit % 2 == 0) {
                result++;
            }
        }
        return result;
    }

}

File: Array/fanbei/0623/c/solution06_23.c
#include <stdio.h>
#include <stdlib.h>

int findNumbers(int *nums, int numsSize);

int findNumbers(int *nums, int numsSize) {
    if (numsSize == 0) {
        return 0;
    }

    int result = 0, tmp = 0, bit = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 0) {
            continue;
        }
        tmp = nums[i];
        bit = 0;
        while (tmp != 0) {
            tmp = tmp / 10;
            bit++;
        }
        if (bit % 2 == 0) {
            result++;
        }
    }

    return result;
}